src/essence/Tools/Isochrone/IsochroneTool.js

```diff
@@ -0,0 +1,539 @@
+
+//Legacy color mapping function
+function hueMap(val) {
+    const hueToChannel = h => {
+        h = (h + 1530) % 1530;
+        if(h < 255) return h;
+        if(h < 765) return 255;
+        if(h < 1020) return 255 - (h % 255);
+        return 0;
+    }
+    const hue = Math.floor(val * 1050);
+    const r = hueToChannel(hue + 510);
+    const g = hueToChannel(hue);
+    const b = hueToChannel(hue - 510);
+    return [r, g, b];
+}
+
+//https://leafletjs.com/reference-1.7.1.html#gridlayer
+L.IsochroneLayer = L.GridLayer.extend({
+    //Override to make layer accept Bounds in place of LatLngBounds
+    //  (cleaner, supports polar projections better)
+    _isValidTile(coords) {
+        const bounds = this.options.bounds;
+        return coords.x >= bounds.min.x
+            && coords.y >= bounds.min.y
+            && coords.x < bounds.max.x
+            && coords.y < bounds.max.y;
+    },
+    createTile: function(coords) {
+        const tile = L.DomUtil.create("canvas", "leaflet-tile");
+
+        const size = this.getTileSize();
+        tile.width = size.x;
+        tile.height = size.y;
+
+        const ctx = tile.getContext("2d");
+        const img = ctx.getImageData(0, 0, size.x, size.y);
+
+        const bounds = this.options.bounds;
+        const tXOffset = coords.x - bounds.min.x;
+        const tYOffset = coords.y - bounds.min.y;
+
+        const alpha = Math.floor(this.options.opacity * 255);
+
+        let di = 0; //img data index
+        for(let y = 0; y < size.y; y++) {
+            const yIndex = tYOffset * size.y + y;
+            for(let x = 0; x < size.x; x++) {
+                const xIndex = tXOffset * size.x + x;
+                const currentData = this.options.data[yIndex][xIndex];
+                if(isFinite(currentData) && currentData < this.options.maxCost) {
+                    const color = IsochroneTool.valueToColor(
+                        currentData / this.options.maxCost,
+                        this.options.color,
+                        this.options.steps
+                    );
+                    img.data[di] = color[0];
+                    img.data[di + 1] = color[1];
+                    img.data[di + 2] = color[2];
+                    img.data[di + 3] = alpha;
+                } else {
+                    img.data[di] = 0;
+                    img.data[di + 1] = 0;
+                    img.data[di + 2] = 0;
+                    img.data[di + 3] = 0;
+                }
+                di += 4;
+            }
+        }
+        ctx.putImageData(img, 0, 0);
+        return tile;
+    }
+});
+
+function backlinkToMove(link) {
+    return [
+        [0, 1],
+        [1, 2],
+        [1, 1],
+        [2, 1],
+        [1, 0],
+        [2, -1],
+        [1, -1],
+        [1, -2],
+        [0, -1],
+        [-1, -2],
+        [-1, -1],
+        [-2, -1],
+        [-1, 0],
+        [-2, 1],
+        [-1, 1],
+        [-1, 2]
+    ][link - 1];
+}
+
+/////////////////////////////////////////////////
```

Comment:
what's this? if you're separating logical section of a file, it's a good idea to leave a comment explaining the difference between them. I would generally avoid using comments as plain decoration